app/controllers/BaseController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.data
+import com.google.inject.Inject
+import logging.Logging
+import models.ValidationError
+import models.helper.Journey
+import models.requests.DataRequest
+import play.api.i18n.I18nSupport
+import play.api.mvc.{AnyContent, Call, Result}
+import services.DataCleansingService
+import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+trait BaseController extends FrontendBaseController with I18nSupport with Logging {
+
+  @Inject var dataCleansingService: DataCleansingService = _
```

Comment:
What is this doing / it looks scary :D

Reply:
It's a field injector, needs to be var because it can't be injected at initialisation. I can't see any way to inject directly to the trait other than this. Saves having to inject on every class that requires it (it's used one of the logErrorsAndContinue methods of the trait).

If you know of a better way, do tell! I'll look now.

Reply:
Tried a few things and failed :( 

Can we just pass in the dataCleansingService to that version of the logErrorsAndContinue? It's not used in many places as we usually don't

Idk, just something seems off to me about trying to get the BaseController to do it. Happy to be wrong though if other people know better :)

Reply:
Currently occupied, but yeah sure we can just pass it in or something similar. I'll take a look tomorrow!

Reply:
Made it implicit rather than injected. Seems like a good middle ground

Reply:
@Ewan-James-Donovan  @ChrisCookOC  In Scala 2 you won't be able to have parameters in a trait, though that is possible in Scala 3. You can use a class if you need that though. However in this case I believe the controller should be doing the cleansing and shouldn't be passed to the base controller as it hides functionality

Reply:
@liandra-giacomine I've done as you suggested

Reply:
yeah makes sense to have it separate :)

Reply:
I've just fixed some bits :)